### PR TITLE
Surrounding F#+ files with compiler directives

### PR DESCRIPTION
**Pull request by jbotto94**
Surrounding all of the F#+ with compiler directives in a similar way as was done with `Externals.fs` in #197 and #160.

### CODE REVIEW

**Comment by gusty** [comment]
I think that in order to add consistency to this PR, code that uses those excluded functions should also be surrounded, otherwise this PR would break Fable compilation.

**Comment by wallymathieu** [comment]
Should the js files be included?

**Comment by jbotto94** [comment]
> Should the js files be included?

They shouldn't. Those are local build files. I have fixed this problem.
This PR should now only alter the core F#+ files, .gitignore and package.json inside FSharpPlusFable.Tests.

File: src/FSharpPlus/Converter.fs [hunk 1]
```diff
@@ -106,28 +123,35 @@ type TryParse with
...
 #if NET35
 #else
+#if !FABLE_COMPILER
     static member        Parse (_: 'T when 'T : enum<_>, _: Parse   ) = fun x ->
         (match Enum.TryParse (x) with
             | (true, v) -> v
             | _         -> invalidArg "value" ("Requested value '" + x + "' was not found.")
         ) : 'enum
 #endif
+#endif
```
Comment (wallymathieu) [inline src/FSharpPlus/Converter.fs]: double #endif?

File: src/FSharpPlus/Functor.fs [hunk 2]
```diff
@@ -23,7 +23,9 @@ type Bind =
     static member        (>>=) (source: seq<'T>    , f: 'T -> seq<'U>     ) = Seq.bind f source                                             : seq<'U> 
 #if NET35
 #else
+#if !FABLE_COMPILER
     static member        (>>=) (source: Task<'T>   , f: 'T -> Task<'U>    ) = source.ContinueWith(fun (x: Task<_>) -> f x.Result).Unwrap () : Task<'U>
+#endif
```
Comment (wallymathieu) [inline src/FSharpPlus/Functor.fs]: double #endif?

File: .gitignore [hunk 3]
```diff
@@ -202,3 +202,8 @@ docsrc/tools/FSharp.Formatting.svclog
 # Node install directory
 
 tests/FSharpPlusFable.Tests/node_modules/
+
+
+# Fable compilation files
+
+tests/FSharpPlusFable.Tests/FSharpPlusFable
```
Comment (wallymathieu) [inline .gitignore]: shouldn't this be more specific to exclude only js files? You could add a .gitignore to that folder to exclude the compilation output.

File: src/FSharpPlus/Converter.fs (hunk 1 above)
Comment (jbotto94) [reply src/FSharpPlus/Converter.fs]: If you see there are two compiler directives:
`#if NET35 ... #else ... #endif` and `#if !FABLE_COMPILER ... #endif`
The double `#endif` is for each `#if`.

File: tests/FSharpPlusFable.Tests/package.json [hunk 4]
```diff
@@ -1,6 +1,7 @@
 {
   "private": true,
   "scripts": {
+    "buildFSharpPlus": "fable-splitter ../../src/FSharpPlus -o FSharpPlusFable --commonjs --allFiles",
```
Comment (jbotto94) [inline tests/FSharpPlusFable.Tests/package.json]: @wallymathieu This is the local target I've added to the local build file of the F#+ Fable tests.

File: .gitignore (hunk 3 above)
Comment (jbotto94) [reply .gitignore]: The folder FSharpPlusFable is created when you compile F#+ with Fable through the target I've added. I've tagged you at this line. This makes git ignore any result of compiling F#+ locally.

**Comment by wallymathieu** [comment]
What do you think @gusty, should we merge this one?

**Comment by gusty** [comment]
Ideally we would have another reviewer from the web side.
Since nothing has happened since, I will approve it and we can merge it.
If someone finds a problem with this, we would have to fix it.
CC: @jannesiera

**Pull request merged**